connector/src/main/java/org/jboss/as/connector/deployers/ds/processors/DriverProcessor.java

```diff
@@ -76,7 +76,7 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
                                 Integer.valueOf(minorVersion));
                     }
                     String driverName = deploymentUnit.getName();
-                    if (! deploymentUnit.getName().endsWith(".jar") || driverNames.size() != 1) {
+                    if ((deploymentUnit.getName().contains(".") && ! deploymentUnit.getName().endsWith(".jar")) || driverNames.size() != 1) {
```

Comment:
I'm not really sure if this is a good idea. It seems very odd that this behaviour is determined by the presence of a '.' in the deployment name, although I can't really think of any other way for the user to specify the driver name.